### PR TITLE
CC-808 | SFUI | SFAutocompleteChip component 

### PR DESCRIPTION
**Pull request by Yopo-Nicolas**

### CODE REVIEW

File: src/Components/SFChipsListField/SFChipsListField.tsx [hunk 1]
```diff
@@ -1,79 +1,18 @@
...
-  }
-});
+import { makeStyles } from '@material-ui/core/styles';
+import { StyledAutoComplete } from '../SFAutocompleteChips/SFAutocompleteChip';
+import {
+  minWidthInputSize,
+  SFAutocompleteInput
+} from '../SFAutocompleteChips/SFAutocompleteInput/SFAutocompleteInput';
 
 const chipsDisplay = makeStyles({
```
Comment (martinweingart) [inline src/Components/SFChipsListField/SFChipsListField.tsx:17]: rename to chipsDisplayClasses

**Review by martinweingart — CHANGES_REQUESTED** [review]
Need changes

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  title: 'Components/SFAutocompleteChip',
+  component: SFAutocompleteChip,
+  parameters: { controls: { sort: 'alpha' } },
+  args: {
+    label: 'Bagel',
+    value: []
+  },
+  argTypes: {
+    disabled: {
+      description: 'If true, the chip should be displayed in a disabled state.'
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx]: Changed to component

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+} as Meta;
+
+const AutocompleteChipItems = [
+  'This War of Mine',
+  'Back to the Future',
+  'Toy Story',
+  'Pulp Fiction',
+  'WALL·E',
+  'The Godfather'
+];
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx]: Was done to reutilize items in multiple stories, changed for only one story

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const Template: Story<SFAutocompleteChipProps> = (args) => {
+  const [values, setValues] = React.useState<string[]>(args.value);
+
+  return <SFAutocompleteChip {...args} value={values} onChange={setValues} />;
+};
+export const Default = Template.bind({});
+
+Default.args = {
+  label: 'Movies',
+  emptyMessage: 'No movies saved...',
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.stories.tsx]: Removed

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  inputMinWidth?: minWidthInputSize;
+  label: string;
+  helperText?: string;
+  options?: string[];
+  disabled?: boolean;
+  required?: boolean;
+  onChange: (newValue: string[]) => void;
+}
+
+export const SFAutocompleteChip = ({
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx]: Changed parent folder name

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  helperText,
+  options = [],
+  value = [],
+  disabled = false,
+  required = false,
+  onChange
+}: SFAutocompleteChipProps): React.ReactElement<SFAutocompleteChipProps> => {
+  const [inputValue, setInputValue] = React.useState<string>('');
+
+  const deleteValue = (input: string): void => {
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx]: Renamed

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    onChange(values);
+  };
+
+  const isValueAlreadyAdded = (input: string): boolean => {
+    const matchValueItem: string | undefined = value.find(
+      (item) => item.toLowerCase() === input.toLowerCase()
+    );
+
+    return matchValueItem !== undefined;
+  };
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx]: Removed

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+        if (
+          valueTrim !== '' &&
+          !isValueAlreadyAdded(valueTrim) &&
+          getValueFromOptions(valueTrim)
+        ) {
+          setInputValue('');
+          onChange([...value, input]);
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx]: Removed unused code

File: src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        value={value}
+        inputValue={inputValue}
+        onInputChange={onInputChange}
+        onChange={onAutoCompleteChange}
+        filterSelectedOptions
+        getOptionSelected={(option: string, value: string): boolean =>
+          option === value
+        }
+        renderTags={(value: string[]): JSX.Element => (
+          <SFAutocompleteChipRender values={value} onDelete={deleteValue} />
```
[the remark this answers is not in the document]
Comment (Yopo-Nicolas) [reply src/Components/SFAutocompleteChips/SFAutocompleteChip.tsx]: done

**Review by martinweingart — APPROVED** [review]
LGTM

**Review by pdcastano — CHANGES_REQUESTED** [review]
Disable = true allows remove chips options

**Comment by pdcastano** [comment]
LGTM

**Review by pdcastano — APPROVED** [review]

**Pull request merged**